### PR TITLE
Add castxml

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
arch: https://aur.archlinux.org/packages/castxml/
debain: https://packages.debian.org/buster/castxml
fedora: https://fedora.pkgs.org/31/fedora-updates-x86_64/castxml-0.3.6-1.fc31.x86_64.rpm.html
gentoo: https://gpo.zugaina.org/dev-cpp/CastXML/RDep
macports: https://ports.macports.org/port/castxml/summary
ubuntu: https://packages.ubuntu.com/bionic/castxml

### CODE REVIEW

**Comment by clalancette** [comment]
Can you please add links to the package database for the Linux distribution packages?

**Comment by MatthijsBurgh** [comment]
@clalancette done

File: rosdep/base.yaml [hunk 1]
```diff
@@ -388,6 +388,13 @@ can-utils:
   debian: [can-utils]
   fedora: [can-utils]
   ubuntu: [can-utils]
+castxml:
+  arch: [castxml]
+  debian: [castxml]
+  fedora: [castxml]
+  gentoo: [dev-cpp/castxml]
```
Comment (clalancette) [inline rosdep/base.yaml:395]: The link you provided for gentoo isn't the normal place we look for gentoo packages.  Running this [search](https://packages.gentoo.org/packages/search?q=castxml) yields no results for gentoo, so I'd remove it.  The rest look OK to me.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml (hunk 1 above)
Comment (MatthijsBurgh) [reply rosdep/base.yaml]: Done

**Review by clalancette — APPROVED** [review]

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**